logstash-core/src/main/java/org/logstash/instrument/metrics/FlowCapture.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.logstash.instrument.metrics;
+
+import java.math.BigDecimal;
+
+/**
+ * A {@link FlowCapture} is used by {@link FlowMetric} to hold
+ * point-in-time data for a pair of {@link Metric}s.
+ * It is immutable.
+ */
+class FlowCapture {
+    private final Number numerator;
+    private final Number denominator;
```

Comment:
_Future note: I have [prototyped using primitive types](https://github.com/yaauie/logstash/commit/flow-metrics-optimized-captures) to optimize the `FlowCapture` to reduce memory usage, but determined it to be out of scope of this PR._